src/cx_ed25519.c

```diff
@@ -0,0 +1,190 @@
+/* https://ed25519.cr.yp.to/python/ed25519.py */
+#include <stdbool.h>
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+
+#include "cx_ed25519.h"
+
+static const char *constant_q = "57896044618658097711785492504343953926634992332820282019728792003956564819949";
+static bool initialized;
+static BIGNUM *d1, *d2, *q, *q_minus_two, *two;
+static const BIGNUM *one;
+static BN_CTX *ctx;
+
+typedef struct {
+  BIGNUM *x;
+  BIGNUM *y;
+} POINT;
+
+static void inv(BIGNUM *r, BIGNUM *x)
+{
+  BN_mod_inverse(r, x, q, ctx);
```

Comment:
It might make the code easier to read if this function is inlined in its callers